ale_linters/scala/metals.vim

```diff
@@ -33,7 +39,21 @@ function! ale_linters#scala#metals#GetProjectRoot(buffer) abort
         endif
     endfor
 
-    return ''
+    for l:root in l:potential_project_root_dirs
+        let l:project_root = ale#path#FindNearestDirectory(
+        \   a:buffer,
+        \   l:root,
+        \)
+
+        if !empty(l:project_root)
+            return fnamemodify(
+            \   l:project_root,
+            \   ':p:h:h',
+            \)
+        endif
+    endfor
+
+    return expand('#' . a:buffer . ':p:h')
```

Comment:
As I would love to be able to use Metals on standalone files, as a compromise, I'm thinking perhaps we can let the user configure the root detection behaviour.  Maybe something like `let g:ale_scala_metals_always_start = v:true` to make sure Metals would start at the current file if no other roots are found.

Reply:
By default we do not want ALE to start spawining language servers for every file open as this can cause performance issues to users. Anyone that prefers to change this behavior can do so by setting the g:ale_root variable, no need to add additional configuration options. The following example shows how to set g:ale_root to return the current buffer path as root if one is not found:

```vimscript
function MyFindMetalRoot(buffer)

  let l:root = ale_linters#scala#metals#GetProjectRoot(a:buffer)

  if !empty(l:root)
      return l:root
  endif

  return expand('#' . a:buffer . ':p:h')
endfunction

let g:ale_root = {
      \ 'metals': function('MyFindMetalRoot')
\}
```